src/main/java/se/diabol/jenkins/pipeline/token/TokenUtils.java

```diff
@@ -0,0 +1,50 @@
+/*
+This file is part of Delivery Pipeline Plugin.
+
+Delivery Pipeline Plugin is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Delivery Pipeline Plugin is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Delivery Pipeline Plugin.
+If not, see <http://www.gnu.org/licenses/>.
+*/
+package se.diabol.jenkins.pipeline.token;
+
+import java.io.IOException;
+
+import org.jenkinsci.plugins.tokenmacro.MacroEvaluationException;
+import org.jenkinsci.plugins.tokenmacro.TokenMacro;
+
+import hudson.model.TaskListener;
+import hudson.model.AbstractBuild;
+
+public final class TokenUtils {
+
+    /**
+     * Expand all tokens for provided template via TokenMacro.expandAll
+     * @param build - current build
+     * @param template - template to decode
+     * @return decoded template or empty when an error occurred or build is empty
+     */
+    public final static String decodedTemplate(AbstractBuild<?, ?> build, String template) {
+        try {
+            if (build != null) {
+                return TokenMacro.expandAll(build, TaskListener.NULL, template);
+            }
+        } catch (MacroEvaluationException e) {
+            e.printStackTrace();
```

Comment:
printStackTrace directs output to the console and should be replaced with logger debugging